core/src/main/java/brooklyn/plan/PlanToSpecFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package brooklyn.plan;
+
+import java.util.Collection;
+import java.util.ServiceLoader;
+
+import com.google.common.collect.Lists;
+
+import brooklyn.management.ManagementContext;
+
+public class PlanToSpecFactory {
+    public static PlanToSpecTransformer forMime(ManagementContext mgmt, String mime) {
+        ServiceLoader<PlanToSpecTransformer> loader = ServiceLoader.load(PlanToSpecTransformer.class);
+        for (PlanToSpecTransformer transformer : loader) {
+            transformer.injectManagementContext(mgmt);
+            if (transformer.accepts(mime)) {
+                return transformer;
+            }
+        }
+        return null;
```

Comment:
I was wondering if I should add this method at all, currently not used. Makes sense to throw if a specific implementation is not found, will change it.